최창익/app/src/main/java/com/mash_up/mvvmstudy/repository/MainRepository.kt

```diff
@@ -0,0 +1,37 @@
+package com.mash_up.mvvmstudy.repository
+
+import com.mash_up.mvvmstudy.repository.model.Repositories
+import com.mash_up.mvvmstudy.repository.remote.ClientFactory
+import com.mash_up.mvvmstudy.repository.remote.GitService
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainRepository {
+    private val api: GitService = ClientFactory.createService(GitService::class.java)
+
+    fun getRepositories(
+        query: String,
+        onSuccess: (Repositories) -> Unit,
+        onError: (String) -> Unit
+    ) {
+        api.getRepositories(query).enqueue(object : Callback<Repositories> {
+            override fun onResponse(call: Call<Repositories>, response: Response<Repositories>) {
+                if (response.isSuccessful && response.body() != null) {
+                    onSuccess(response.body()!!)
```

Comment:
변수에 한번 저장해서 `!!` 키워드 안쓰는건 어떤가요!!
```suggestion
                val date = response.body()
                if (response.isSuccessful && date != null) {
                    onSuccess(date)
```

Reply:
오 좋아여 `!!`는 최대한 안쓰는게 좋을거같은데 반영해볼게여!